agent/xds/clusters.go

```diff
@@ -111,7 +112,7 @@ func (s *Server) clustersFromSnapshotMeshGateway(cfgSnap *proxycfg.ConfigSnapsho
 	// generate the service subset clusters
 	for svc, resolver := range cfgSnap.MeshGateway.ServiceResolvers {
 		for subsetName, _ := range resolver.Subsets {
-			clusterName := ServiceSNI(svc, subsetName, "default", cfgSnap.Datacenter, cfgSnap)
+			clusterName := connect.ServiceSNI(svc, subsetName, "default", cfgSnap.Datacenter, cfgSnap.Roots.TrustDomain)
```

Comment:
Would it be better to take this from the API (new `Name` field) instead of keeping the duplicated logic? I think that should be available form the target here right?

I don't think it matters too much for now if it's hard for some reason but it seems to be bypassing the new source of truth being the API rather than just shared conventions?

Reply:
This is the mesh gateway branch of the xds code that doesn't talk to the discovery chain.